include/ofi_util.h

```diff
@@ -1370,6 +1370,14 @@ static inline void ofi_cq_err_memcpy(uint32_t api_version,
 	}
 }
 
+static inline
+bool ofi_lockless_progress(enum fi_threading threading, enum fi_progress control)
+{
+	return (threading == FI_THREAD_DOMAIN || threading == FI_THREAD_COMPLETION) &&
+		control == FI_PROGRESS_CONTROL_UNIFIED ?
+			true : false;
+}
+
```

Comment:
The double negative in the name here is confusing. You also use it in the exact same way every time (ie find the bool and then pick noop vs mutex. I think it would make more sense to combine and simplify:
```
static enum ofi_lock_type ofi_progress_lock_type(enum fi_threading threading, enum fi_progress control)
{
	return (threading == FI_THREAD_DOMAIN || threading == FI_THREAD_COMPLETION) &&
		control == FI_PROGRESS_CONTROL_UNIFIED ? OFI_LOCK_NOOP : OFI_LOCK_MUTEX;
}
```

Reply:
done